### PR TITLE
implement basic types

### PR DESCRIPTION
**Pull request by merceroncode**
### Metanorma PR checklist

 - [ ] Breaking changes (list related PRs)
 - [x] Documentation update required ([create task for this](https://github.com/metanorma/metanorma.org/issues/new))
 - [x] External dependency introduced ([documentation update need](https://github.com/metanorma/metanorma.org/issues/new))
 - [ ] Gem with native library introduced

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,3 +10,5 @@ gem "rake", "~> 13.0"
 gem "rspec", "~> 3.0"
 
 gem "rubocop", "~> 1.21"
+
+gem "shale"
```
Comment (ronaldtse) [inline Gemfile:14]: This should go into edoxen.gemspec

File: Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,59 @@
+PATH
```
Comment (ronaldtse) [inline Gemfile.lock:1]: Remove Gemfile.lock from Git.

File: lib/edoxen/action.rb [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+    ALLOWED_VALUES = ['adopts', 'thanks', 'approves', 'decides', 'declares', 'asks', 'invites', 'resolves', 'confirms',
+    'welcomes', 'recommends', 'requests', 'congratulates', 'instructs', 'urges', 'appoints', 'resolves further',
+    'instructs', 'calls-upon', 'encourages', 'affirms', 'elects', 'authorizes', 'charges', 'states', 'remarks', 'judges',
+    'sanctions', 'abrogates', 'empowers'].freeze
+
+    def self.cast(value)
+      if ALLOWED_VALUES.include?(value.to_s)
+        value.to_sym
+      else
+        raise ConsiderationTypeEnumInvalidValue, "#{value.inspect} is not a valid action type"
```
Comment (ronaldtse) [inline lib/edoxen/action.rb:18]: Rewrite this as:

```ruby
def self.cast(value)
  unless ALLOWED_VALUES.include?(value.to_s)
    raise ConsiderationTypeEnumInvalidValue, "#{value.inspect} is not a valid action type"
  end

  value.to_sym
end
```

File: lib/edoxen/approval.rb [hunk 4]
```diff
@@ -15,3 +15,43 @@
...
+module Edoxen
+  class ConsiderationTypeEnumInvalidValue < TypeError; end
+
+  class ApprovalDegreeEnumInvalidValue < TypeError; end
+
+  class ConsiderationTypeEnum < Shale::Type::Value
+    ALLOWED_VALUES = ['affirmative', 'negative'].freeze
+
+    def self.cast(value)
+      if ALLOWED_VALUES.include?(value.to_s)
```
Comment (ronaldtse) [inline lib/edoxen/approval.rb:29]: As above.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

**Comment by merceroncode** [comment]
@ronaldtse Hey, did you hear anything from the other Marcin? I didn't hear from him for a month, that's actually why no work has been done.

File: lib/edoxen/approval.rb (hunk 4 above)
Comment (merceroncode) [reply lib/edoxen/approval.rb:29]: Hey, so this was corrected almost as soon as you've said this. But as I've said, I kind of can't contact Marcin

@webdev778 what about you man? I'm sorry we didn't get along the first time around, but I'm getting really worried about him

**Comment by merceroncode** [comment]
OK, sorry, communication restored

**Comment by ronaldtse** [comment]
Finally merging. Thank you @merceroncode ! We now have lutaml-model that replaces Shale!

**Pull request merged**